### ISSUE TITLE
Date Range Not working

### ISSUE THREAD

**Issue by payros** [payros/mmda#1]
(@jarai): Inputted a link today (12/7) and then set my date range to end on 12/5, but the link from 12/7 is still showing up.

**Comment by payros** [comment]
I tested this and found another (unrelated) bug where the *default* checkbox is not unchecked when you check a non-default box. Fixed it on [this commit](https://github.com/payros/mmda/commit/ccf20fb8f14553a2b883bb0d6e7a552aad2122ea)

I couldn't reproduce your issue. Are you selecting the "insert date" or the "reference date" filter?

If you're referring to the "reference date", I found several bugs with it and fixed them on [this commit](https://github.com/payros/mmda/commit/e8f9cb15911a23286ad9cb77a24fe23a1c3ce679)

If not, let me know so I can investigate some more.
